### PR TITLE
fix: Mandatory VRF request fees estimation

### PR DESCRIPTION
**Pull request by akhercha**
Closes #NA

## Introduced changes

- Re-added mandatory fee check
- Multicall impl (without wait_for_tx)
- whitelist for which we don't check fees

Note: Adding `track_nonce` adds way to much unreliability. Better to loose a few seconds waiting for the tx than pilling up txs that will fail and retry requests.

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: pragma-sdk/pragma_sdk/onchain/mixins/randomness.py [hunk 1]
```diff
@@ -509,14 +477,16 @@ async def handle_random(
...
         invoke_tx = await self.submit_random_multicall(vrf_submit_requests)
         if not invoke_tx:
             logger.error(f"⛔ VRF Submission for {len(events)} failed!")
         else:
             logger.info(
                 f"✅ Submitted the VRF responses for {len(events)} requests:"
-                f" {hex(invoke_tx.transaction_hash)}"
+                f" {hex(invoke_tx.hash)}"
             )
+        print(f"Send tx took: {(time.time()) - NOW:02f}s")
```
Comment (EvolveArt) [inline pragma-sdk/pragma_sdk/onchain/mixins/randomness.py:489]: same

File: vrf-listener/benchmark/stress/stress_tester.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ async def run_stresstest(self) -> None:
         print("\n🧩 Creating user clients...")
         users = await self.config.get_users_client(
             network=self.network,
-            rpc_url=self.rpc_url,
+            rpc_url="https://api.cartridge.gg/x/starknet/sepolia",
```
Comment (EvolveArt) [inline vrf-listener/benchmark/stress/stress_tester.py:71]: revert this change ? rpc should be configurable no?

Comment (akhercha) [reply vrf-listener/benchmark/stress/stress_tester.py]: done

File: pragma-sdk/pragma_sdk/onchain/mixins/randomness.py (hunk 1 above)
Comment (akhercha) [reply pragma-sdk/pragma_sdk/onchain/mixins/randomness.py]: done

File: pragma-sdk/pragma_sdk/onchain/mixins/randomness.py [hunk 3]
```diff
@@ -166,50 +164,20 @@ async def submit_random_multicall(
...
-                self._get_submit_or_refund_calls(
-                    request=request,
-                    check_fees=False,
-                )
+                self._get_submit_or_refund_calls(request=request)
                 for request in vrf_requests
             ]
         )
         all_calls = flatten_list(all_calls)
+        print(f"Call creation took: {(time.time()) - ESS:02f}s")
```
[the remark this answers is not in the document]
Comment (akhercha) [reply pragma-sdk/pragma_sdk/onchain/mixins/randomness.py]: done

**Comment by akhercha** [comment]
@EvolveArt 

Also added a `whitelist` param to the vrf-listener.
We can now add whitelisted addresses for which we won't check fees & pay if necessary for the callback.

Also added in the `infra` - just need to add an env variable `LOOT_CONTRACT_ADDRESS`

**Review by EvolveArt — APPROVED** [review]
